python/test/unit/language/test_line_info.py

```diff
@@ -212,3 +212,33 @@ def test_line_info_interpreter(func: str):
         expected_def_lineno = 68
     kernel.rewrite()
     assert kernel.rewriter.def_file_lineno == expected_def_lineno
+
+
+@pytest.mark.parametrize("status", ["0", "1"])
+def test_line_info_env(status: str):
+    if is_interpreter():
+        pytest.skip("interpreter does not support warmup compilation")
+
+    try:
+        obj_kind, command, anchor, separator = get_disassembler_command_and_debug_line_format()
+    except BaseException:
+        pytest.skip("disassembler is not available")
+
+    import contextlib
+
+    @contextlib.contextmanager
+    def disable_line_info(status: str):
+        import os
+        original_status = os.environ.get("TRITON_DISABLE_LINE_INFO", "0")
+        try:
+            os.environ["TRITON_DISABLE_LINE_INFO"] = status
+            yield
+        finally:
+            os.environ["TRITON_DISABLE_LINE_INFO"] = original_status
+
+    shape = (128, )
+    with disable_line_info(status):
+        kernel_single.device_caches.clear()
+        kernel_info = kernel_single.warmup(torch.float32, torch.float32, BLOCK=shape[0], grid=(1, ))
```

Comment:
NIT: Could be simplified with the [`monkeypatch` fixture](https://docs.pytest.org/en/stable/how-to/monkeypatch.html#monkeypatching)
```suggestion
    monkeypatch.setenv("TRITON_DISABLE_LINE_INFO", status)
    kernel_single.device_caches.clear()
    kernel_info = kernel_single.warmup(torch.float32, torch.float32, BLOCK=shape[0], grid=(1, ))
```

Reply:
Sure, this is handy!